google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/CloudDebugSettingsEditor.java

```diff
@@ -30,14 +34,23 @@
 
   private final CloudDebugRunConfigurationPanel settingsPanel;
 
-  public CloudDebugSettingsEditor() {
+  CloudDebugSettingsEditor() {
     settingsPanel = new CloudDebugRunConfigurationPanel();
+    settingsPanel
+        .getProjectSelector()
+        .addProjectSelectionListener(
+            (selectedProject) -> {
+              // settings editor does not see all the changes by default, use explicit notification.
+              settingsPanel.triggerValidation();
+            });
   }
 
   @Override
-  protected void applyEditorTo(CloudDebugRunConfiguration runConfiguration)
-      throws ConfigurationException {
-    runConfiguration.setCloudProjectName(settingsPanel.getProjectName());
+  protected void applyEditorTo(@NotNull CloudDebugRunConfiguration runConfiguration) {
+    runConfiguration.setCloudProjectName(
+        Optional.ofNullable(settingsPanel.getProjectSelector().getSelectedProject())
```

Comment:
do you think it would be better separation to have the settings panel return the CloudProject instead of the selector widget itself?

Reply:
yes, probably cleaner, I'll change that.